### PR TITLE
Extend garbage collection documentation.

### PR DESCRIPTION
**Pull request by RichardScothern**
Signed-off-by: Richard Scothern <richard.scothern@docker.com>

### CODE REVIEW

**Comment by RichardScothern** [comment]
ping @dmcgowan

File: docs/garbage-collection.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+If a layer is deleted it will be removed from the filesystem when garbage collection
+is run.  If a manifest is deleted the layers to which it refers will be removed from
+the filesystem if no other manifests refers to them.
+
+
+### Example
+
+In this example manifest A references two layers: `a` and `b`.  Manifest `B` references
+layers `b` and `c`.  In this state, nothing is eligible for garbage collection:
```
Comment (dmcgowan) [inline docs/garbage-collection.md]: diagram shows `a` and `c`

File: docs/garbage-collection.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+registry versions the intention is that garbage collection will be an automated background
+action and this manual process will no longer apply.
+
+
+
+# Running garbage collection
+
+You can run garbage collection by running
+
+`docker run --rm <registry-image-name> garbage-collect /path/to/config.yml`
```
Comment (dmcgowan) [inline docs/garbage-collection.md]: Can we provide instructions for running this outside a container as well. For an example command seems lacking some needed details such as mounting volumes, stopping registry container or putting it in read-only mode.

**Comment by dmcgowan** [comment]
LGTM

**Pull request merged**